### DISCUSSION TITLE
snapshot that contains full contents of a file

### DISCUSSION THREAD

**Discussion by wholtz** [askimed/nf-test#252] [Questions] [answered]
I have some small text files that I am creating snapshots of. I would prefer that the snapshot store the full content of these files rather than just the md5 sum so that I can get diffs when changes are found. Is this currently supported?

Thanks -- love the tool!

**Comment by lukfor** [comment, marked as answer]
When using snapshot assertions to compare the contents of files, the `text` property is used to directly access the textual content of the files. This allows you to compare the actual text within the files rather than just comparing their md5 sum.

Example:
```groovy
assert snapshot(
    path("${params.outdir}/file1.txt").text, 
    path("${params.outdir}/file2.txt").text
).match()
```

**Reply by wholtz** [reply]
Oh right -- so simple. Thanks!
